src/meta/src/hummock/manager/mod.rs

```diff
@@ -1714,11 +1708,11 @@ where
         {
             self.check_state_consistency().await;
         }
-        Ok(())
+        Ok(Some(snapshot))
```

Comment:
We may return an error in L1704 and thus miss to notify frontend. Can this be an issue? Recovery should be triggered in this case so maybe missing one notification is fine?

Reply:
Refactored `try_send_compaction_request` to make it infallible (by handling errors internally).
Because I think a compaction scheduling failure should not trigger recovery. And when it fails, recovery doesn't help at all. And currently the only reason of failure is compaction scheduler shutdown.